haskell/src/Monocle/Backend/Queries.hs

```diff
@@ -287,6 +288,32 @@ getProjectAgg query = do
         )
       ]
 
+-- | The repos_summary query
+getTermKey :: BH.TermsResult -> Text
+getTermKey (BH.TermsResult (BH.TextValue tv) _ _) = tv
+getTermKey BH.TermsResult {} = error "Unexpected match"
+
+getTermsAgg :: BH.Query -> Text -> TenantM [BH.TermsResult]
+getTermsAgg query onTerm = do
+  search <- aggSearch query aggs
+  pure $ filter isNotEmptyTerm $ unfilteredR search
+  where
+    aggs = BH.mkAggregations "singleTermAgg" $ BH.TermsAgg $ BH.mkTermsAggregation onTerm
+    toTermsResult bucket = BH.buckets bucket
+    unfilteredR search' = maybe [] toTermsResult (BH.toTerms "singleTermAgg" search')
```

Comment:
```suggestion
    unfilteredR search' = maybe [] BH.buckets (BH.toTerms "singleTermAgg" search')
```

Reply:
And using eta-reduction: `unfilteredR = maybe [] BH.buckets . BH.toTerms "singleTermAgg"`